jpo-ode-svcs/src/main/java/us/dot/its/jpo/ode/services/asn1/Asn1EncodedDataRouter.java

```diff
@@ -90,26 +98,28 @@ public Object process(String consumedData) {
 
          if (metadata.has(TimTransmogrifier.REQUEST_STRING)) {
             JSONObject request = metadata.getJSONObject(TimTransmogrifier.REQUEST_STRING);
-
             if (request.has(TimTransmogrifier.RSUS_STRING)) {
-               JSONObject rsusIn = (JSONObject) request.get(TimTransmogrifier.RSUS_STRING);
-               if (rsusIn.has(TimTransmogrifier.RSUS_STRING)) {
-                  Object rsu = rsusIn.get(TimTransmogrifier.RSUS_STRING);
-                  JSONArray rsusOut = new JSONArray();
-                  if (rsu instanceof JSONArray) {
-                     logger.debug("Multiple RSUs exist in the request: {}", request);
-                     JSONArray rsusInArray = (JSONArray) rsu;
-                     for (int i = 0; i < rsusInArray.length(); i++) {
-                        rsusOut.put(rsusInArray.get(i));
+               Object rsus = request.get(TimTransmogrifier.RSUS_STRING);
+               if (rsus instanceof JSONObject) {
+                  JSONObject rsusIn = (JSONObject) request.get(TimTransmogrifier.RSUS_STRING);
+                  if (rsusIn.has(TimTransmogrifier.RSUS_STRING)) {
+                     Object rsu = rsusIn.get(TimTransmogrifier.RSUS_STRING);
+                     JSONArray rsusOut = new JSONArray();
+                     if (rsu instanceof JSONArray) {
```

Comment:
I think it would improve readability and comprehension for the reader if we extracted the blocks of these "if" statements into methods with descriptive names. Also, while it might be less impactful, extracting the conditions being checked into methods with descriptive names could also improve clarity. We should consider implementing these changes at some point.